src/ray/protobuf/usage.proto

```diff
@@ -94,4 +94,37 @@ enum TagKey {
   NUM_ACTOR_TASKS = 306;
   NUM_NORMAL_TASKS = 307;
   NUM_DRIVERS = 308;
+
+  // Data
+  // Read operators.
+  DATA_READ_PARQUET_OP = 400;
```

Comment:
We shouldn't add these many tags. Can you instead put the operations inside the value? We can accumulate used operations in a comma-separated list and record on execution if it has changed.

E.g., `DATA_OPS_TAG -> {"parquet_read", "map", "flat_map"}`

Reply:
I hate these many tags as well and happy to reduce as many as possible. Given current telemetry implementation, the key-value pair is buffered in GCS memory, for every one hour per cluster. So if multiple jobs run within the time window, the last job's `DATA_OPS_TAG` would overwrite the previous jobs.

Looks like this is solvable, to hold the lock of `_recorded_extra_usage_tags_lock`, and update the key `DATA_OPS_TAG` in-place to append the value (basically `_recorded_extra_usage_tags.get()` followed by `_recorded_extra_usage_tags.put()`), cc @jjyao.

Reply:
I think it's fine for now to allow one job to overwrite another: statistically you still know the ops usage.

Reply:
Updated to track the operator names and count in dictionary.